File: rules-cli/src/main/java/FilterListener.java
import io.github.iamrenny.ruleflow.RuleFlowLanguageParser;
import org.antlr.v4.runtime.TokenStream;

public class FilterListener extends PrettyPrinter {
    private final TokenStream tokens;
    private final String criteria;

    public FilterListener(TokenStream tokens, String criteria) {
        super(tokens);
        this.tokens = tokens;
        this.criteria = criteria;
    }

    @Override
    public void enterRules(RuleFlowLanguageParser.RulesContext ctx) {
        if (tokens.getText(ctx.getStart(), ctx.getStop()).contains(criteria)) {
            super.enterRules(ctx);
        }
    }

    @Override
    public void exitRules(RuleFlowLanguageParser.RulesContext ctx) {
        if (tokens.getText(ctx.getStart(), ctx.getStop()).contains(criteria)) {
            super.exitRules(ctx);
        }
    }
}

File: rules-cli/src/main/kotlin/FilterListener.kt
import com.github.iamrenny.ruleflow.RuleFlowLanguageBaseListener
import com.github.iamrenny.ruleflow.RuleFlowLanguageParser
import org.antlr.v4.runtime.TokenStream

class FilterListener(tokens: TokenStream, val criteria: String): PrettyPrinter(tokens) {
    override fun enterRules(ctx: RuleFlowLanguageParser.RulesContext) {
        if(tokens.getText(ctx.start, ctx.stop).contains(criteria)) {
            super.enterRules(ctx)
        }
    }

    override fun exitRules(ctx: RuleFlowLanguageParser.RulesContext) {
        if (tokens.getText(ctx.start, ctx.stop).contains(criteria)) {
            super.exitRules(ctx)
        }
    }
}